plugins/org.jboss.tools.openshift.common.core/src/org/jboss/tools/openshift/common/core/utils/URIUtils.java

```diff
@@ -0,0 +1,46 @@
+/*******************************************************************************
+ * Copyright (c) 2015 Red Hat, Inc. Distributed under license by Red Hat, Inc.
+ * All rights reserved. This program is made available under the terms of the
+ * Eclipse Public License v1.0 which accompanies this distribution, and is
+ * available at http://www.eclipse.org/legal/epl-v10.html
+ * 
+ * Contributors: Red Hat, Inc.
+ ******************************************************************************/
+package org.jboss.tools.openshift.common.core.utils;
+
+import java.util.HashMap;
+import java.util.Map;
+import java.net.URI;
+import java.net.URISyntaxException;
+
+/**
+ * Helper methods for manipulating URIs
+ */
+public class URIUtils {
+	
+	private URIUtils(){
+	}
+	
+	public static  Map<String, String> splitFragment(String location){
+		URI uri = null;
+		try {
+			uri = new URI(location);
+		} catch (URISyntaxException e) {
+			throw new RuntimeException(e);
+		}
+		return splitFragment(uri);
+	}
+	
+	public static Map<String, String> splitFragment(URI uri){
```

Comment:
@adietish they are similiar to URL parameters but there is no '?' to indicate the query values so they aren't returned from the URL when you make that call, but they are returned as part of the fragment